File: 0034-find-first-and-last-position-of-element-in-sorted-array/0034-find-first-and-last-position-of-element-in-sorted-array.c
/**
 * Note: The returned array must be malloced, assume caller calls free().
 */
int* searchRange(int* nums, int numsSize, int target, int* returnSize) 
{
    int *p = (int*)malloc(2*sizeof(int));
    *returnSize=2;
    p[0]=-1;
    p[1]=-1;
    for(int i=0; i<numsSize; i++)
    {
        if(nums[i]==target)
        {
            p[0]=i;
            break;
        }
    }    
    for(int i=numsSize-1; i>=0; i--)
    {
        if(nums[i]==target)
        {
            p[1]=i;
            break;
        }
    }
    return p;
}

File: 0034-find-first-and-last-position-of-element-in-sorted-array/0034-find-first-and-last-position-of-element-in-sorted-array.java
class Solution {
    public int[] searchRange(int[] nums, int target) {
        int[] ans = new int[2];

        int n = nums.length;
        ans[0] = -1;
        ans[1] = -1;
        
        for(int i=0; i<n; i++)
        {
            if(nums[i] == target)
            {
                ans[0] = i;
                break;
            }
        }
        for(int j=n-1; j>=0; j--)
        {
            if(nums[j] == target)
            {
                ans[1] = j;
                break;
            }
        }
        return ans;
    }
}